sdk/python/lib/pulumi/__init__.py

```diff
@@ -97,3 +97,86 @@
 )
 
 from . import runtime, dynamic, policy, x
+
+__all__ = [
+    # asset
+    "Asset",
+    "Archive",
+    "AssetArchive",
+    "FileArchive",
+    "FileAsset",
+    "RemoteArchive",
+    "RemoteAsset",
+    "StringAsset",
+
+    # config
+    "Config",
+    "ConfigMissingError",
+    "ConfigTypeError",
+
+    # errors
+    "RunError",
+
+    # invoke
+    "InvokeOptions",
+
+    # metadata
+    "get_project",
+    "get_stack",
+
+    # resource
+    "Alias",
+    "Resource",
+    "CustomResource",
+    "CustomTimeouts",
+    "ComponentResource",
+    "ProviderResource",
+    "ResourceOptions",
+    "create_urn",
+    "export",
+    "ROOT_STACK_RESOURCE",
+    "ResourceTransformation",
+    "ResourceTransformationArgs",
+    "ResourceTransformationResult",
+
+    # output
+    "Output",
+    "Input",
+    "Inputs",
+    "InputType",
+    "UNKNOWN",
+    "contains_unknowns",
+
+    # log
+    "debug",
+    "info",
+    "warn",
+    "error",
+
+    # stack_reference
+    "StackReference",
+
+    # _types
+    "MISSING",
+    "input_type",
+    "output_type",
+    "property",
+    "getter",
+    "get",
+    "set",
+
+    # sub-modules
+    "asset",
+    "config",
+    "errors",
+    "invoke",
+    "metadata",
+    "resource",
+    "output",
+    "log",
+    "stack_reference",
+    "runtime",
+    "dynamic",
+    "policy",
+    "x"
```

Comment:
Should we prune some of these submodules that aren't really intended to be used as modules -- the ones where we're already importing from them and exporting at the top-level?

```suggestion
    "runtime",
    "dynamic",
    "policy",
    "x"
```

Might require having to fix-up some imports internally or in tests.

Reply:
Yeah, I was wondering this myself and hoping you'd confirm this for me 😄👍

Reply:
Although, it looks like VSCode still has file names show up in autocomplete (as evidenced by the screenshot of the `After` automation autocomplete. Seems like if it doesn't start with an `_` it will show up in autocomplete. Wondering if 3.0 might be an opportunity to rename the files that we don't intend on exporting.

Reply:
For `automation` can we go ahead and rename such files to have the preceding `_`? (Definitely before we promote from `x`.)

Reply:
Yup, doable for `automation` even now since it's in preview and all of the documentation/examples import directly from `automation` anyway.